public/app/plugins/panel/candlestick/fields.ts

```diff
@@ -180,5 +188,48 @@ export function prepareCandlestickFields(
     }
   }
 
+  // Make sure first field is time!
+  if (!options.includeAllFields) {
+    const fields: Field[] = [frame.fields[timeIndex]];
+    for (const f of frame.fields) {
+      if (used.has(f)) {
+        fields.push(f);
+      }
+    }
+    data.frame = {
+      ...data.frame,
+      fields,
+    };
+  } else if (timeIndex > 0) {
+    const fields = frame.fields.slice();
+    const tmp = fields[0];
+    fields[0] = frame.fields[timeIndex];
+    fields[timeIndex] = tmp;
+    data.frame = {
+      ...data.frame,
+      fields,
+    };
+  }
+
+  // Force update all the indicies
+  for (let i = 0; i < data.frame.fields.length; i++) {
```

Comment:
@leeoniya -- not sure if anything from line 203-233 is necessary, this was my attempt to fix things not showing up at all :shrug: